lang/csharp/src/apache/main/Schema/MapSchema.cs

```diff
@@ -68,8 +68,9 @@ internal static MapSchema NewInstance(JToken jtok, PropertyMap props, SchemaName
         /// Constructor for map schema class
         /// </summary>
         /// <param name="valueSchema">schema for map values type</param>
-        /// <param name="props">dictionary that provides access to custom properties</param>
-        private MapSchema(Schema valueSchema, PropertyMap props) : base(Type.Map, props)
+        /// <param name="cutsomProperties">dictionary that provides access to custom properties</param>
+        public MapSchema(Schema valueSchema, PropertyMap cutsomProperties = null)
```

Comment:
Schema classes already use a Factory pattern method "NewInstance". I would recommend following the same pattern than exposing the constructor.